components/nexus-httpclient/src/main/java/org/sonatype/nexus/httpclient/config/ProxyConfiguration.java

```diff
@@ -63,6 +65,25 @@ public void setNonProxyHosts(@Nullable final String[] nonProxyHosts) {
     this.nonProxyHosts = nonProxyHosts;
   }
 
+  public ProxyConfiguration copy() {
+    try {
+      ProxyConfiguration copy = (ProxyConfiguration)clone();
+      if (http != null) {
+        copy.http = http.copy();
+      }
+      if (https != null) {
+        copy.https = https.copy();
+      }
+      if (nonProxyHosts != null) {
+        copy.nonProxyHosts = Arrays.copyOf(nonProxyHosts, nonProxyHosts.length);
```

Comment:
good to know, will update.  This a String[] so nothing deeper needed in this case.